tests/bluetooth/bsim_bt/bsim_test_mesh/src/settings_test_backend.c

```diff
@@ -54,8 +60,8 @@ static ssize_t settings_line_read_cb(void *cb_arg, void *data, size_t len)
 {
 	struct line_read_ctx *valctx = (struct line_read_ctx *)cb_arg;
 
-	if ((valctx->len / 2) != len) {
-		return 0;
+	if ((valctx->len / 2) > len) {
+		return -1;
```

Comment:
I added this comment in Pavel's PR as well, but the for loop and return below must use `valctx->len` now. Also, please use the errno return codes. This should be `-ENOBUFS` or `-ENOMEM`. These functions will only be used in testing, so it's not the end of the world, but it helps with readability.

Reply:
rebased on main with merged Pavel's PR